tests/test_drain.py

```diff
@@ -73,4 +73,43 @@ def test_add_log_message_sim_75(self):
             actual.append(cluster.get_template())
 
         self.assertListEqual(list(map(str.strip, expected)), actual)
-        self.assertEqual(8, model.get_total_cluster_size())
\ No newline at end of file
+        self.assertEqual(8, model.get_total_cluster_size())
+
+    def test_max_size(self):
```

Comment:
please also add a test comparing Drain with unlimited cluster count to Drain with LRU of practically unlimited size. Results shall be the same.
Also I think that testing on very few log messages may not be good enough. Its better to test on the full input of the big file for that purpose.

Reply:
I confirmed this change doesn't change the logic by inspecting templates produced by the `examples/drain_bigfile_demo.py` script.